Core.DomainServices/Repositories/GDPR/DataProcessingRegistrationReadModelRepository.cs

```diff
@@ -68,5 +70,24 @@ public IQueryable<DataProcessingRegistrationReadModel> GetByUserId(int userId)
                 .Where(x => x.RoleAssignments.Any(assignment => assignment.UserId == userId || x.LastChangedById == userId))
                 .Distinct();
         }
+
+        public IQueryable<DataProcessingRegistrationReadModel> GetReadModelsMustUpdateToChangeActiveState()
+        {
+
+            var now = DateTime.Now;
+            var expiringReadModelIds = _repository
+                .AsQueryable()
+                .Transform(new QueryReadModelsWhichShouldExpire(now).Apply)
+                .Select(x => x.Id);
+
+            var activatingReadModelIds = _repository
+                .AsQueryable()
+                .Transform(new QueryReadModelsWhichShouldBecomeActive(now).Apply)
+                .Select(x => x.Id);
+
+            var idsOfReadModelsWhichMustUpdate = expiringReadModelIds.Concat(activatingReadModelIds).Distinct().ToList();
+
+            return _repository.AsQueryable().ByIds(idsOfReadModelsWhichMustUpdate);
+        }
```

Comment:
Also make sure you update this one in the same file to take the main contract into account:

![image](https://user-images.githubusercontent.com/13116889/208374479-76ca9e83-2a35-4177-8b48-7e59e752576e.png)